![Logo](reports/figures/Screenshot%202022-07-29%20at%204.59.17%20PM.png)

<style>
/* Header 2 */
h2 {
  font-family: "Bahnschrift", sans-serif;
  font-size: 28px;
}

</style>

## Introduction

The dataset gives information about the confirmed cases in UK. The dataset contains daily data from Jan to November 2020.

In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import datetime as dt
import seaborn as sns
import plotly.graph_objects as go
import plotly.colors as colors
import plotly.express as px


In [3]:
# Importing the Data and Presenting the data

df = pd.read_csv('data/raw/specimenDate_ageDemographic-unstacked.csv')
df.head()

areaType   areaCode    areaName        date  newCasesBySpecimenDate-0_4  \
0     ltla  E06000001  Hartlepool  2020-03-16                           0   
1     ltla  E06000001  Hartlepool  2020-03-17                           0   
2     ltla  E06000001  Hartlepool  2020-03-18                           0   
3     ltla  E06000001  Hartlepool  2020-03-19                           0   
4     ltla  E06000001  Hartlepool  2020-03-20                           0   

   newCasesBySpecimenDate-0_59  newCasesBySpecimenDate-10_14  \
0                            0                             0   
1                            0                             0   
2                            0                             0   
3                            0                             0   
4                            0                             0   

   newCasesBySpecimenDate-15_19  newCasesBySpecimenDate-20_24  \
0                             0                             0   
1                             0                             0   
2                             0                             0   
3                             0                             0   
4                             0                             0   

   newCasesBySpecimenDate-25_29  ...  newCasesBySpecimenDateRollingSum-5_9  \
0                             0  ...                                     0   
1                             0  ...                                     0   
2                             0  ...                                     0   
3                             0  ...                                     0   
4                             0  ...                                     0   

   newCasesBySpecimenDateRollingSum-60+  \
0                                     0   
1                                     0   
2                                     0   
3                                     0   
4                                     0   

   newCasesBySpecimenDateRollingSum-60_64  \
0                                       0   
1                                       0   
2                                       0   
3                                       0   
4                                       0   

   newCasesBySpecimenDateRollingSum-65_69  \
0                                       0   
1                                       0   
2                                       0   
3                                       0   
4                                       0   

   newCasesBySpecimenDateRollingSum-70_74  \
0                                       0   
1                                       0   
2                                       0   
3                                       0   
4                                       0   

   newCasesBySpecimenDateRollingSum-75_79  \
0                                       0   
1                                       0   
2                                       0   
3                                       0   
4                                       0   

   newCasesBySpecimenDateRollingSum-80_84  \
0                                       0   
1                                       0   
2                                       0   
3                                       0   
4                                       0   

   newCasesBySpecimenDateRollingSum-85_89  \
0                                       0   
1                                       0   
2                                       0   
3                                       0   
4                                       0   

   newCasesBySpecimenDateRollingSum-90+  \
0                                     0   
1                                     0   
2                                     0   
3                                     0   
4                                     0   

   newCasesBySpecimenDateRollingSum-unassigned  
0                                            0  
1                                            0  
2  

In [4]:
df.areaType.unique()

array(['ltla', 'nation', 'overview', 'region', 'utla'], dtype=object)

# Exploratory Data Analysis
##### Exploring data and Checking any missing Values

In [5]:
df.columns # Columns in the dataset 

Index(['areaType', 'areaCode', 'areaName', 'date',
       'newCasesBySpecimenDate-0_4', 'newCasesBySpecimenDate-0_59',
       'newCasesBySpecimenDate-10_14', 'newCasesBySpecimenDate-15_19',
       'newCasesBySpecimenDate-20_24', 'newCasesBySpecimenDate-25_29',
       'newCasesBySpecimenDate-30_34', 'newCasesBySpecimenDate-35_39',
       'newCasesBySpecimenDate-40_44', 'newCasesBySpecimenDate-45_49',
       'newCasesBySpecimenDate-50_54', 'newCasesBySpecimenDate-55_59',
       'newCasesBySpecimenDate-5_9', 'newCasesBySpecimenDate-60+',
       'newCasesBySpecimenDate-60_64', 'newCasesBySpecimenDate-65_69',
       'newCasesBySpecimenDate-70_74', 'newCasesBySpecimenDate-75_79',
       'newCasesBySpecimenDate-80_84', 'newCasesBySpecimenDate-85_89',
       'newCasesBySpecimenDate-90+', 'newCasesBySpecimenDate-unassigned',
       'newCasesBySpecimenDateRollingRate-0_4',
       'newCasesBySpecimenDateRollingRate-0_59',
       'newCasesBySpecimenDateRollingRate-10_14',
       'newCasesBySpecime

In [6]:
df['areaType'].unique()

array(['ltla', 'nation', 'overview', 'region', 'utla'], dtype=object)

In [7]:
df['areaCode'].unique()

array(['E06000001', 'E06000002', 'E06000003', 'E06000004', 'E06000005',
       'E06000006', 'E06000007', 'E06000008', 'E06000009', 'E06000010',
       'E06000011', 'E06000012', 'E06000013', 'E06000014', 'E06000015',
       'E06000016', 'E06000017', 'E06000018', 'E06000019', 'E06000020',
       'E06000021', 'E06000022', 'E06000023', 'E06000024', 'E06000025',
       'E06000026', 'E06000027', 'E06000030', 'E06000031', 'E06000032',
       'E06000033', 'E06000034', 'E06000035', 'E06000036', 'E06000037',
       'E06000038', 'E06000039', 'E06000040', 'E06000041', 'E06000042',
       'E06000043', 'E06000044', 'E06000045', 'E06000046', 'E06000047',
       'E06000049', 'E06000050', 'E06000051', 'E06000052', 'E06000054',
       'E06000055', 'E06000056', 'E06000057', 'E06000058', 'E06000059',
       'E07000004', 'E07000005', 'E07000006', 'E07000007', 'E07000008',
       'E07000010', 'E07000011', 'E07000012', 'E07000026', 'E07000027',
       'E07000028', 'E07000029', 'E07000030', 'E07000031', 'E070

In [8]:
df['areaName'].unique()

array(['Hartlepool', 'Middlesbrough', 'Redcar and Cleveland',
       'Stockton-on-Tees', 'Darlington', 'Halton', 'Warrington',
       'Blackburn with Darwen', 'Blackpool',
       'Kingston upon Hull, City of', 'East Riding of Yorkshire',
       'North East Lincolnshire', 'North Lincolnshire', 'York', 'Derby',
       'Leicester', 'Rutland', 'Nottingham', 'Herefordshire, County of',
       'Telford and Wrekin', 'Stoke-on-Trent',
       'Bath and North East Somerset', 'Bristol, City of',
       'North Somerset', 'South Gloucestershire', 'Plymouth', 'Torbay',
       'Swindon', 'Peterborough', 'Luton', 'Southend-on-Sea', 'Thurrock',
       'Medway', 'Bracknell Forest', 'West Berkshire', 'Reading',
       'Slough', 'Windsor and Maidenhead', 'Wokingham', 'Milton Keynes',
       'Brighton and Hove', 'Portsmouth', 'Southampton', 'Isle of Wight',
       'County Durham', 'Cheshire East', 'Cheshire West and Chester',
       'Shropshire', 'Cornwall and Isles of Scilly', 'Wiltshire',
       'Bedford

In [9]:
# Finding any null Values present in the dataset

df.info() 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 114800 entries, 0 to 114799
Data columns (total 69 columns):
 #   Column                                       Non-Null Count   Dtype  
---  ------                                       --------------   -----  
 0   areaType                                     114800 non-null  object 
 1   areaCode                                     114800 non-null  object 
 2   areaName                                     114800 non-null  object 
 3   date                                         114800 non-null  object 
 4   newCasesBySpecimenDate-0_4                   114800 non-null  int64  
 5   newCasesBySpecimenDate-0_59                  114800 non-null  int64  
 6   newCasesBySpecimenDate-10_14                 114800 non-null  int64  
 7   newCasesBySpecimenDate-15_19                 114800 non-null  int64  
 8   newCasesBySpecimenDate-20_24                 114800 non-null  int64  
 9   newCasesBySpecimenDate-25_29                 114800 non-nul

#### As there no null values in the dataset we can proced futher but the date is the type Object we need to change it into Datetime

# Checking & Cleaning Data

In [10]:
# Chaging the 'date' Column to datetime format so we can use date properties during Analysis

df['date'] = pd.to_datetime(df['date'])
df['date'].head()

0   2020-03-16
1   2020-03-17
2   2020-03-18
3   2020-03-19
4   2020-03-20
Name: date, dtype: datetime64[ns]

In [11]:
# Checking whether all the Columns in required Dtype

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 114800 entries, 0 to 114799
Data columns (total 69 columns):
 #   Column                                       Non-Null Count   Dtype         
---  ------                                       --------------   -----         
 0   areaType                                     114800 non-null  object        
 1   areaCode                                     114800 non-null  object        
 2   areaName                                     114800 non-null  object        
 3   date                                         114800 non-null  datetime64[ns]
 4   newCasesBySpecimenDate-0_4                   114800 non-null  int64         
 5   newCasesBySpecimenDate-0_59                  114800 non-null  int64         
 6   newCasesBySpecimenDate-10_14                 114800 non-null  int64         
 7   newCasesBySpecimenDate-15_19                 114800 non-null  int64         
 8   newCasesBySpecimenDate-20_24                 114800 non-null  in

In [12]:
# Checking that sum in the respective specimens is equal to the other individual five year bands

Cases_of_ageGroup_0_59 = ['newCasesBySpecimenDate-0_4','newCasesBySpecimenDate-10_14', 
                          'newCasesBySpecimenDate-15_19','newCasesBySpecimenDate-20_24', 
                          'newCasesBySpecimenDate-25_29','newCasesBySpecimenDate-30_34', 
                          'newCasesBySpecimenDate-35_39','newCasesBySpecimenDate-40_44', 
                          'newCasesBySpecimenDate-45_49','newCasesBySpecimenDate-50_54',
                          'newCasesBySpecimenDate-55_59','newCasesBySpecimenDate-5_9']

df['cases_of_ageGroup_0to59'] = df[Cases_of_ageGroup_0_59].sum(axis=1)
df[['newCasesBySpecimenDate-0_59','cases_of_ageGroup_0to59']]


newCasesBySpecimenDate-0_59  cases_of_ageGroup_0to59
0                                 0                        0
1                                 0                        0
2                                 0                        0
3                                 0                        0
4                                 0                        0
...                             ...                      ...
114795                          136                      133
114796                          137                      134
114797                          129                      128
114798                          108                      108
114799                           90                       89

[114800 rows x 2 columns]

In [13]:
# Checking Whether the individual five year bands upto 59 are equal with the column "newCasesBySpecimenDate-0_59"

check = df['newCasesBySpecimenDate-0_59'] == df['cases_of_ageGroup_0to59']
check.describe()

count     114800
unique         2
top         True
freq      109682
dtype: object

#### As we see from the data only 109682 rows are matching with the data given but there are about 5118 rows are not matching with the data given

In [14]:
# New list for 0_59

Cases_of_ageGroup_0_59_new = ['newCasesBySpecimenDate-0_4','newCasesBySpecimenDate-10_14', 
                              'newCasesBySpecimenDate-15_19','newCasesBySpecimenDate-20_24', 
                              'newCasesBySpecimenDate-25_29','newCasesBySpecimenDate-30_34', 
                              'newCasesBySpecimenDate-35_39','newCasesBySpecimenDate-40_44', 
                              'newCasesBySpecimenDate-45_49','newCasesBySpecimenDate-50_54',
                              'newCasesBySpecimenDate-55_59','newCasesBySpecimenDate-5_9',
                              'newCasesBySpecimenDate-unassigned']

df['cases_of_ageGroup_0to59_new'] = df[Cases_of_ageGroup_0_59_new].sum(axis=1)
df[['newCasesBySpecimenDate-0_59','cases_of_ageGroup_0to59','cases_of_ageGroup_0to59_new']]


newCasesBySpecimenDate-0_59  cases_of_ageGroup_0to59  \
0                                 0                        0   
1                                 0                        0   
2                                 0                        0   
3                                 0                        0   
4                                 0                        0   
...                             ...                      ...   
114795                          136                      133   
114796                          137                      134   
114797                          129                      128   
114798                          108                      108   
114799                           90                       89   

        cases_of_ageGroup_0to59_new  
0                                 0  
1                                 0  
2                                 0  
3                                 0  
4                                 0  
...                             ...  
114795                          136  
114796                          137  
114797                          129  
114798                          108  
114799                           90  

[114800 rows x 3 columns]

#### After Exploring data throughly there is a column "newCasesBySpecimenDate-unassigned " adding this  to the individual five year bands upto 59. Then 'newCasesBySpecimenDate-0_59' is matching with the sum.




In [15]:
# Checking Whether the individual five year bands upto 59 and Unassigned are equal with the column "newCasesBySpecimenDate-0_59"

check = df['newCasesBySpecimenDate-0_59'] == df['cases_of_ageGroup_0to59_new']
check.describe()


count     114800
unique         1
top         True
freq      114800
dtype: object

In [16]:
# Similarly checking individual five year bands after 60 are equal with the column "newCasesBySpecimenDate-60+"

Cases_of_ageGroup_above_60 = ['newCasesBySpecimenDate-60_64', 'newCasesBySpecimenDate-65_69',
                              'newCasesBySpecimenDate-70_74', 'newCasesBySpecimenDate-75_79',
                              'newCasesBySpecimenDate-80_84','newCasesBySpecimenDate-85_89',
                              'newCasesBySpecimenDate-90+']

df['Cases_of_ageGroup_above_60'] = df[Cases_of_ageGroup_above_60].sum(axis=1)
df[['newCasesBySpecimenDate-60+','Cases_of_ageGroup_above_60']]


newCasesBySpecimenDate-60+  Cases_of_ageGroup_above_60
0                                1                           1
1                                0                           0
2                                1                           1
3                                0                           0
4                                1                           1
...                            ...                         ...
114795                          35                          35
114796                          25                          25
114797                          27                          27
114798                          17                          17
114799                          19                          19

[114800 rows x 2 columns]

In [17]:

Check2 = df['newCasesBySpecimenDate-60+'] == df['Cases_of_ageGroup_above_60']
Check2.describe()


count     114800
unique         1
top         True
freq      114800
dtype: object

### Conclusion:
#### From the Checking & Cleaning data the main outcomes are 
1. newCasesByspecimenDate-0_59 = "sum of individual five year bands upto 59 "+"newCasesBySpecimenDate-unassigned    
2. newCasesByspecimenDate-60+  = "sum of individual five year bands after age 60 " 
3. Total_cases = newCasesByspecimenDate-0_59 + newCasesByspecimenDate-60+ 


# Creating New Columns

In analyzing data and creating graphs according to our needs we need to create new columns for Total cases
and Rolling sum for Week days and creating a new data set with only these columns save us a huge amount of compilation time and increase efficiency.

In [18]:
Total_cases = ['newCasesBySpecimenDate-0_59','newCasesBySpecimenDate-60+']
df['Total_cases'] = df[Total_cases].sum(axis=1)

In [19]:
df

areaType   areaCode        areaName       date  \
0          ltla  E06000001      Hartlepool 2020-03-16   
1          ltla  E06000001      Hartlepool 2020-03-17   
2          ltla  E06000001      Hartlepool 2020-03-18   
3          ltla  E06000001      Hartlepool 2020-03-19   
4          ltla  E06000001      Hartlepool 2020-03-20   
...         ...        ...             ...        ...   
114795     utla  E10000034  Worcestershire 2020-10-28   
114796     utla  E10000034  Worcestershire 2020-10-29   
114797     utla  E10000034  Worcestershire 2020-10-30   
114798     utla  E10000034  Worcestershire 2020-10-31   
114799     utla  E10000034  Worcestershire 2020-11-01   

        newCasesBySpecimenDate-0_4  newCasesBySpecimenDate-0_59  \
0                                0                            0   
1                                0                            0   
2                                0                            0   
3                                0                            0   
4                                0                            0   
...                            ...                          ...   
114795                           2                          136   
114796                           4                          137   
114797                           4                          129   
114798                           6                          108   
114799                           3                           90   

        newCasesBySpecimenDate-10_14  newCasesBySpecimenDate-15_19  \
0                                  0                             0   
1                                  0                             0   
2                                  0                             0   
3                                  0                             0   
4                                  0                             0   
...                              ...                           ...   
114795                             3                            23   
114796                             6                             6   
114797                             6                            12   
114798                             7                            14   
114799                             3                             5   

        newCasesBySpecimenDate-20_24  newCasesBySpecimenDate-25_29  ...  \
0                                  0                             0  ...   
1                                  0                             0  ...   
2                                  0                             0  ...   
3                                  0                             0  ...   
4                                  0                             0  ...   
...                              ...                           ...  ...   
114795                            12                            15  ...   
114796                            12                            13  ...   
114797                            16                            10  ...   
114798                             8                            13  ...   
114799                             8                             7  ...   

        newCasesBySpecimenDateRollingSum-70_74  \
0                                            0   
1                                            0   
2                                            0   
3                                            0   
4                                            0   
...                                        ...   
114795                                      24   
114796                                      27   
114797                                      28   
114798                                      32   
114799                                      33   

        newCasesBySpecimenDateRollingSum-75_79  \
0                                            0   
1                                            0   
2                       

#### Rolling Sum & Rolling Mean
Rolling window statistics are very frequently used in analyzing and smoothing time-series data.

In [20]:
# here we are defining rolling sum and rolling average over a period of 7 days

df['Running_Sum'] = df['Total_cases'].rolling(7).sum()
df['Running_Average'] = df['Total_cases'].rolling(7).mean()


#### Cumulative Sum

In [21]:
# Cumulative sum is accumulation of totalCases over a period of time

df['Cumulative_Sum'] = df['Total_cases'].cumsum()


In [22]:
df_new = df[['areaName','date','Total_cases','Running_Sum','Running_Average','Cumulative_Sum']]
df_new


areaName       date  Total_cases  Running_Sum  Running_Average  \
0           Hartlepool 2020-03-16            1          NaN              NaN   
1           Hartlepool 2020-03-17            0          NaN              NaN   
2           Hartlepool 2020-03-18            1          NaN              NaN   
3           Hartlepool 2020-03-19            0          NaN              NaN   
4           Hartlepool 2020-03-20            1          NaN              NaN   
...                ...        ...          ...          ...              ...   
114795  Worcestershire 2020-10-28          171        905.0       129.285714   
114796  Worcestershire 2020-10-29          162        923.0       131.857143   
114797  Worcestershire 2020-10-30          156        965.0       137.857143   
114798  Worcestershire 2020-10-31          125       1017.0       145.285714   
114799  Worcestershire 2020-11-01          109       1040.0       148.571429   

        Cumulative_Sum  
0                    1  
1                    1  
2                    2  
3                    2  
4                    3  
...                ...  
114795         4534050  
114796         4534212  
114797         4534368  
114798         4534493  
114799         4534602  

[114800 rows x 6 columns]

#  Analysis Part

### Groupby Date

In [23]:
# Groupby date i.e finding Cases for each day

df_date = df[['date','Total_cases']]
date_group = df_date.groupby(pd.Grouper(key='date', axis=0, freq='D')).sum()
date_group = date_group.reset_index()
date_group

date  Total_cases
0   2020-01-30            7
1   2020-01-31            0
2   2020-02-01            0
3   2020-02-02            5
4   2020-02-03            0
..         ...          ...
272 2020-10-28       100494
273 2020-10-29        99025
274 2020-10-30        97474
275 2020-10-31        70784
276 2020-11-01        65991

[277 rows x 2 columns]

In [24]:

# Groupby areaName i.e finding Cases for each area

fig = go.Figure(data=go.Scatter(x=date_group['date'], y=date_group['Total_cases']))

fig.update_layout(
    template="plotly_dark",
    title_text='Daily Cases',
    title_font_size=25,
    xaxis_rangeslider_visible=True,
    xaxis_rangeslider_bordercolor='black',  # Set the color of the range slider border
    xaxis_title="Date",
    yaxis_title="No. of Cases",


)

fig.show()




In [25]:
# daily change

date_group['daily_change'] = date_group['Total_cases'].diff()
date_group.head()


date  Total_cases  daily_change
0 2020-01-30            7           NaN
1 2020-01-31            0          -7.0
2 2020-02-01            0           0.0
3 2020-02-02            5           5.0
4 2020-02-03            0          -5.0

In [26]:
fig = go.Figure(data=go.Scatter(x=date_group['date'], y=date_group['daily_change']))

fig.update_layout(
    title="Daily Change in Cases",
    xaxis_title="Date",
    yaxis_title="No of Cases",
    template="plotly_dark", 
    title_font_size=25,

)

fig.show()

In [27]:
# Groupby Week i.e finding total cases that are recorded the whole week

weekly_group = df_date.groupby(pd.Grouper(key='date', axis=0, freq='W')).sum()
weekly_group = weekly_group.reset_index()
weekly_group.head()



date  Total_cases
0 2020-02-02           12
1 2020-02-09           35
2 2020-02-16           60
3 2020-02-23           10
4 2020-03-01          267

In [28]:
# plot of date vs total cases in weekly_group database

fig = go.Figure(data=go.Scatter(x=weekly_group['date'], y=weekly_group['Total_cases']))

fig.update_layout(
    
    template="plotly_dark",
    title="weekly Change in Cases",
    title_font_size=25,
    xaxis_rangeslider_visible=True,
    xaxis_rangeslider_bordercolor='black',  # Set the color of the range slider border
    xaxis_title="Date",
    yaxis_title="No. of Cases"

)

fig.show()


In [29]:
# Groupby Month i.e finding total cases that are recorded until the End of the Month

monthly_group = df_date.groupby(pd.Grouper(key='date', axis=0, freq='M')).sum()
monthly_group = monthly_group.reset_index()
monthly_group['Month'] = monthly_group['date'].dt.month_name()
monthly_group = monthly_group[['Month','Total_cases']]
monthly_group.sort_values(by='Total_cases',ascending=False,inplace=True)
monthly_group

Month  Total_cases
9     October      2396165
8   September       632443
3       April       573742
4         May       336183
2       March       160503
7      August       147646
5        June       126169
6        July        95476
10   November        65991
1    February          277
0     January            7

In [30]:
palette = colors.qualitative.Pastel

fig = go.Figure(data=go.Bar(y=monthly_group['Month'], x=monthly_group['Total_cases'], orientation='h',marker=dict(color=palette)))

fig.update_layout(
    template="plotly_dark",
    title="Monthly Cases",
    xaxis_title="No of Cases",
    yaxis_title="Months",
    title_font_size=25,

)

fig.show()

# 

### Groupby Place

In [31]:
# Groupby Place allows us view the each place over the period of time

df_place = df[['areaName','Total_cases']]
place_group = df_place.groupby(['areaName']).sum()
place_group = place_group.reset_index()
place_group.head()

areaName  Total_cases
0          Adur          403
1     Allerdale          963
2  Amber Valley         1863
3          Arun          866
4      Ashfield         2256

In [32]:
# Ten Places with Most Cases

Highest_cases = place_group.sort_values(by='Total_cases').head(10)
Highest_cases



areaName  Total_cases
306       Torridge          191
326     West Devon          246
172         Maldon          342
259     South Hams          399
199  North Norfolk          403
0             Adur          403
233         Rother          410
133       Hastings          412
192    North Devon          432
240        Ryedale          461

In [33]:
# Interactive Graph
palette = colors.qualitative.Pastel

fig = go.Figure(data=go.Bar(y=Highest_cases['areaName'], x=Highest_cases['Total_cases'], orientation='h',marker=dict(color=palette)))

fig.update_layout(
    template="plotly_dark",
    title="Areas with Highest No of Cases",
    xaxis_title="No of Cases",
    yaxis_title="Area Name",
    title_font_size=25,

)

fig.show()

In [34]:
# Ten places with Least Cases

Least_cases = place_group.sort_values(by='Total_cases',ascending=False).head(10)
Least_cases


areaName  Total_cases
101                   England       916619
310            United Kingdom       916619
203                North West       226827
350  Yorkshire and The Humber       139584
169                    London       103646
329             West Midlands        96766
89              East Midlands        85872
257                South East        80871
193                North East        64373
95            East of England        56603

In [35]:
# Interactive Graph

palette = colors.qualitative.Pastel

fig = go.Figure(data=go.Bar(y=Least_cases['areaName'], x=Least_cases['Total_cases'], orientation='h',marker=dict(color=palette)))

fig.update_layout(
    template="plotly_dark",
    title="Areas with Least No of Cases",
    xaxis_title="No of Cases",
    yaxis_title="Area Name",
    title_font_size=25,

)

fig.show()

# 

## Cases on a Particular date

In [36]:
# Using User input 

x = input("Enter the date to find out Total Cases on that praticular day eg:'2020-01-30'")
date_group[date_group['date']==x]


date  Total_cases  daily_change
0 2020-01-30            7           NaN

## Areas with Highest Cases on a given Day

In [37]:
# Creating required Dataset

DateAndArea = df[['date','areaName','Total_cases']]
DateAndArea.head()

date    areaName  Total_cases
0 2020-03-16  Hartlepool            1
1 2020-03-17  Hartlepool            0
2 2020-03-18  Hartlepool            1
3 2020-03-19  Hartlepool            0
4 2020-03-20  Hartlepool            1

In [38]:
# Getting data of Cases in required area

required_place = input(" Enter the name of the Place you want to find: ")
DateAndArea_requiredArea = DateAndArea[DateAndArea['areaName'] == required_place]
DateAndArea_requiredArea.sort_values(by='date',ascending=False,inplace=True)


/var/folders/sp/gqk812nj51j9l5hwqrvrfnyr0000gr/T/ipykernel_8944/2951353115.py:5: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



In [39]:
fig = go.Figure(data=go.Scatter(x=DateAndArea_requiredArea['date'], y=DateAndArea_requiredArea['Total_cases']))

fig.update_layout(
    title="Daily Change in Cases",
    xaxis_title="Date",
    yaxis_title="No of Cases",
    template="plotly_dark", 
    title_font_size=25,

)

fig.show()

In [40]:
# Gives Cases on that particular date and in cases in Top 10 corressponding city

x = input("Enter the date to find out Total Cases on that praticular day eg:'2020-01-30'")
CasesOnaDay = DateAndArea[DateAndArea['date']==x]
CasesOnaDay_Top10_Places =  CasesOnaDay.sort_values(by='Total_cases',ascending = False).head(10)
CasesOnaDay_Top10_Places


date                  areaName  Total_cases
75530 2020-01-30                   England            2
75807 2020-01-30            United Kingdom            2
3077  2020-01-30                      York            1
76603 2020-01-30  Yorkshire and The Humber            1
81519 2020-01-30                      York            1

In [52]:
# Plotting Cases on that particular date and in cases in Top 10 corressponding city

fig = go.Figure(data=go.Scatter(x=CasesOnaDay_Top10_Places['areaName'], y=CasesOnaDay_Top10_Places['Total_cases']))

fig.update_layout(
    title="Cases on a Particular Day",
    xaxis_title="Date",
    yaxis_title="No of Cases",
    template="plotly_dark", 
    title_font_size=25,

)

fig.show()



## Areas with largest positive change of cases on Weekly period

In [53]:
weekly_group['Weekly_Change'] = weekly_group['Total_cases'].diff()
weekly_group.head()

date  Total_cases  Weekly_Change
0 2020-02-02           12            NaN
1 2020-02-09           35           23.0
2 2020-02-16           60           25.0
3 2020-02-23           10          -50.0
4 2020-03-01          267          257.0

In [54]:
# Plotting graph for a Cases in required area


fig = go.Figure(data=go.Scatter(x=weekly_group['date'], y=weekly_group['Weekly_Change']))

fig.update_layout(
    title="Weekly Change of Cases",
    xaxis_title="Date",
    yaxis_title="No of Cases",
    template="plotly_dark", 
    title_font_size=25,

)

fig.show()



## 

## Comparing 2 or More areas

#### Selecting Basis ..
Selecting 2 areas with largest no of cases and 2 areas with lowest no of cases

In [55]:
# Creating Dataset

Places =  ['Torridge', 'West Devon', 'Maldon', 'South Hams', 'North Norfolk','Adur'] # , 'Rother', 'Hastings','North Devon','Ryedale' ]
Comparing_dataset = df[['date','areaName','Total_cases']]
Comparing_dataset = Comparing_dataset[Comparing_dataset['areaName'].isin(Places)]
Comparing_dataset


date    areaName  Total_cases
19527 2020-03-03  South Hams            1
19528 2020-03-04  South Hams            0
19529 2020-03-05  South Hams            0
19530 2020-03-06  South Hams            2
19531 2020-03-07  South Hams            0
...          ...         ...          ...
54328 2020-10-28        Adur            6
54329 2020-10-29        Adur           11
54330 2020-10-30        Adur            6
54331 2020-10-31        Adur            2
54332 2020-11-01        Adur            6

[1419 rows x 3 columns]

In [56]:



# Group the data by 'areaName'
groups = Comparing_dataset.groupby('areaName')
fig = go.Figure()

# Iterate over the groups and create separate traces for each area
for name, group in groups:
    fig.add_trace(go.Scatter(
        x=group['date'],
        y=group['Total_cases'],
        mode='lines',
        name=name  # Set the name for the legend
    ))

fig.update_layout(
    title='Cases in the Selected areas over time',
    xaxis_title='Date',
    yaxis_title='No of Cases',
    template='plotly_dark',
    title_font_size=25
)

fig.update_xaxes(rangeslider_visible=True, rangeselector=dict(
    buttons=list([
        dict(count=7, label='1w', step='day', stepmode='backward'),
        dict(count=1, label='1m', step='month', stepmode='backward'),
        dict(count=6, label='6m', step='month', stepmode='backward'),
        dict(count=1, label='YTD', step='year', stepmode='todate'),
        dict(count=1, label='1y', step='year', stepmode='backward'),
        dict(step='all')
    ]),
    font=dict(color='black')  # Set button text color to black
))

fig.show()


### Cases in a Required Time Period

In [57]:
# Getting data for Cases in the given Time period

start_date = input("Enter the Start date Format:'YYYY-MM-DD' ")
End_date = input("Enter the Start date Format:'YYYY-MM-DD' ")
CaseInGiven_DateRange= date_group[(date_group['date']>= start_date) & (date_group['date']<= End_date)]
CaseInGiven_DateRange


date  Total_cases  daily_change
137 2020-06-15         4651         940.0
138 2020-06-16         4898         247.0
139 2020-06-17         4508        -390.0
140 2020-06-18         4672         164.0
141 2020-06-19         4017        -655.0
..         ...          ...           ...
272 2020-10-28       100494       -1269.0
273 2020-10-29        99025       -1469.0
274 2020-10-30        97474       -1551.0
275 2020-10-31        70784      -26690.0
276 2020-11-01        65991       -4793.0

[140 rows x 3 columns]

In [59]:
fig = go.Figure(data=[
    go.Scatter(
        x=CaseInGiven_DateRange['date'],
        y=CaseInGiven_DateRange['Total_cases'],
        mode='lines',
        fill='tozeroy',  # Fill the area below the line
        line=dict(width=2, color='lightseagreen'),  # Set the line properties
    )
])

fig.update_layout(
    title="Cases in the given Time period",
    xaxis_title="Date",
    yaxis_title="No of Cases",
    template="plotly_dark",
    title_font_size=25
)

fig.show()
